src/value_serializer.rs

```diff
@@ -0,0 +1,382 @@
+  this: &mut ValueSerializerDelegate,
+  message: Local<String>,
+) {
+  ValueSerializerDelegateBase::dispatch_mut(this)
+    .throw_data_clone_error(message)
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn v8__ValueSerializer__Delegate__BASE__WriteHostObject(
+  this: &mut ValueSerializerDelegate,
+  isolate: *mut Isolate,
+  object: Local<Object>,
+) -> MaybeBool {
+  let scope = &mut crate::scope::CallbackScope::new(&mut *isolate);
+  MaybeBool::from(
+    ValueSerializerDelegateBase::dispatch_mut(this)
+      .write_host_object(scope, object),
+  )
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn v8__ValueSerializer__Delegate__BASE__GetSharedArrayBufferId(
+  this: &mut ValueSerializerDelegate,
+  isolate: *mut Isolate,
+  shared_array_buffer: Local<SharedArrayBuffer>,
+  clone_id: *mut u32,
+) -> bool {
+  let scope = &mut crate::scope::CallbackScope::new(&mut *isolate);
+  match ValueSerializerDelegateBase::dispatch_mut(this)
+    .get_shared_array_buffer_id(scope, shared_array_buffer)
+  {
+    Some(x) => {
+      *clone_id = x;
+      true
+    }
+    None => false,
+  }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn v8__ValueSerializer__Delegate__BASE__GetWasmModuleTransferId(
+  this: &mut ValueSerializerDelegate,
+  isolate: *mut Isolate,
+  module: Local<WasmModuleObject>,
+  transfer_id: *mut u32,
+) -> bool {
+  let scope = &mut crate::scope::CallbackScope::new(&mut *isolate);
+  match ValueSerializerDelegateBase::dispatch_mut(this)
+    .get_wasm_module_transfer_id(scope, module)
+  {
+    Some(x) => {
+      *transfer_id = x;
+      true
+    }
+    None => false,
+  }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn v8__ValueSerializer__Delegate__BASE__ReallocateBufferMemory(
+  this: &mut ValueSerializerDelegate,
+  old_buffer: *mut c_void,
+  size: usize,
+  actual_size: *mut usize,
+) -> *mut c_void {
+  let old_buffer = if old_buffer.is_null() {
+    None
+  } else {
+    Some(Box::from_raw(std::ptr::slice_from_raw_parts_mut(
+      old_buffer as *mut u8,
+      size,
+    )))
+  };
+  let new_buffer = ValueSerializerDelegateBase::dispatch_mut(this)
+    .reallocate_buffer_memory(old_buffer);
+  (match new_buffer {
+    Some(mut x) => {
+      *actual_size = x.len();
+      x.as_mut_ptr()
+    }
+    None => std::ptr::null::<u8>(),
+  }) as *mut c_void
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn v8__ValueSerializer__Delegate__BASE__FreeBufferMemory(
+  this: &mut ValueSerializerDelegate,
+  buffer: *mut c_void,
+) {
+  let buffer =
+    Box::from_raw(std::ptr::slice_from_raw_parts_mut(buffer as *mut u8, 1));
+  ValueSerializerDelegateBase::dispatch_mut(this).free_buffer_memory(buffer)
+}
+
+extern "C" {
+  fn v8__ValueSerializer__Delegate__BASE__CONSTRUCT(
+    buf: *mut MaybeUninit<ValueSerializerDelegate>,
+  );
+}
+
+pub trait ValueSerializerDelegateImpl {
+  fn base(&self) -> &ValueSerializerDelegateBase;
+  fn base_mut(&mut self) -> &mut ValueSerializerDelegateBase;
+
+  fn throw_data_clone_error(&mut self, message: Local<String>);
```

Comment:
Unfortunately this function will be a bit useless because there's literally nothing the user can do with the message parameter (because we are not getting a `HandleScope` to work with).